components/SpinnerButton/SpinnerButton.tsx

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import type { ViewStyle } from "react-native";
+import { Animated, TouchableOpacity, View } from "react-native";
+import AnimatableView from "../AnimatableView/AnimatableView";
+import AnimatedView from "../AnimatedView/AnimatedView";
+import ChildrenView from "../ChildrenView/ChildrenView";
+import styles from "../styles/SpinnerButtonStyle";
+import { getSpinnerStyle, useAnimatedValues } from "../utils";
+import { spinnerButtonProps } from "./types";
+
+const AnimatedTouchablesOpacity =
+  Animated.createAnimatedComponent(TouchableOpacity);
+
+const SpinnerButton = ({
```

Comment:
Break this component into smaller components

Reply:
All the coding part is done by Rupal I just gave TypeScript Support. If I try to break down the component it will get messed up.

Reply:
As this one is already implemented I just converted types to TSX and give respected types to the components we are not supposed to break down or change any component apart from TS conversion.